Updated ERC-1, ERGW-3. Added ERCON-2

fguerri opened the pull request:
<!-- Thank you for submitting a Pull Request. Please fill out the template below.-->
# Overview/Summary

This PR:
- Updates ERC-1 to improve clarity (removed the "two circuits for **each** ER gateway" which could be misleading).
- Updates ERGW-3 with severity lowered to "Medium", as it is not possible to delete a gateway with configured connetcions. Therefore, it is not possible to accidentally delete a gateway that is routing traffic between Azure and on-prem. Nevertheless, accidental deletion of a gateway can cause a downtime window to las tlonger than expected due to the need to re-preovision it, so it makes sense to keep the recommendation.
- Adds ERCON-2, which recommends adding resource locks to Connection resources with severity "High". Accidental deltion of connections can lead to outages. 

## Related Issues/Work Items
AB#33852
AB#33853

## This PR fixes/adds/changes/removes

1. *Replace me*
2. *Replace me*
3. *Replace me*

### Breaking Changes
NONE

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [NA] Ensured PR tests are passing
- [NA] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries and/or scripts
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

ehaslett reviewed (approved):
LGTM - thanks!

ejhenry commented:
@fguerri - are you able to address the failed check or do you need help with this?

fguerri commented:
Fixed!

From: Eric Henry ***@***.***>
Sent: Tuesday, March 5, 2024 1:45 AM
To: Azure/Azure-Proactive-Resiliency-Library ***@***.***>
Cc: Federico Guerrini ***@***.***>; Mention ***@***.***>
Subject: Re: [Azure/Azure-Proactive-Resiliency-Library] Updated ERC-1, ERGW-3. Added ERCON-2 (PR #306)


@fguerri<https://github.com/fguerri> - are you able to address the failed check or do you need help with this?

-
Reply to this email directly, view it on GitHub<https://github.com/Azure/Azure-Proactive-Resiliency-Library/pull/306#issuecomment-1977740143>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AETKQM2M7SZ54LO6YZUH7ILYWUIQBAVCNFSM6AAAAABEEYWKL6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNZXG42DAMJUGM>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

ejhenry reviewed (approved):
LGTM 👍
